### PR TITLE
RTN16b

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

**Comment by tcard** [comment]
LGTM

**Comment by tcard** [comment]
@ricardopereira Note how `testAttachedChannelFails` fails from time to time with a timeout. I believe that's because it's the first tests that uses the test app, so it's the one issuing the slow POST /apps request. I still think we should increase the timeout at least for the old test suite.

**Comment by ricardopereira** [comment]
@tcard Ok, I agree but we should only increase the timeout for the [app request](https://github.com/ably/ably-ios/blob/c2904d8c836b68d681fc05e27366d7d64b8a93c4/Tests/ARTTestUtil.m#L308) and leave the `testTimeout` used in our tests.

File: Spec/RealtimeClientConnection.swift [hunk 1]
```diff
@@ -1735,6 +1735,24 @@ class RealtimeClientConnection: QuickSpec {
...
+            context("Connection recovery") {
+
+                // RTN16b
+                it("Connection#recoveryKey should be composed with the connection key and latest serial received") {
+                    let options = AblyTests.commonAppSetup()
+                    let client = ARTRealtime(options: options)
+                    defer { client.close() }
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connected) { _ in
+                            expect(client.connection.recoveryKey).to(equal("\(client.connection.key!):\(client.connection.serial)"))
```
Comment (mattheworiordan) [inline Spec/RealtimeClientConnection.swift]: Could we send one message after this to make sure the serial increases?

File: Spec/RealtimeClientConnection.swift [hunk 2]
```diff
@@ -1894,6 +1894,25 @@ class RealtimeClientConnection: QuickSpec {
...
+                    defer { client.close() }
+                    let channel = client.channels.get("test")
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connected) { _ in
+                            expect(client.connection.serial).to(equal(-1))
+                            expect(client.connection.recoveryKey).to(equal("\(client.connection.key!):\(client.connection.serial)"))
+                        }
+                        channel.publish(nil, data: "message") { error in
+                            expect(error).to(beNil())
+                            expect(client.connection.serial).to(equal(0))
```
Comment (ricardopereira) [inline Spec/RealtimeClientConnection.swift]: Done. (I rebase it. It was not intended. Sorry)

File: Spec/RealtimeClientConnection.swift (hunk 1 above)
Comment (ricardopereira) [inline Spec/RealtimeClientConnection.swift]: https://github.com/ably/ably-ios/pull/392#discussion_r61413476

**Pull request merged**